lldb/test/API/functionalities/process_save_core_minidump/TestProcessSaveCoreMinidump.py

```diff
@@ -36,17 +35,37 @@ def verify_core_file(
             self.assertEqual(module_file_name, expected_file_name)
             self.assertEqual(module.GetUUIDString(), expected.GetUUIDString())
 
+        red_zone = process.GetTarget().GetStackRedZoneSize()
         for thread_idx in range(process.GetNumThreads()):
             thread = process.GetThreadAtIndex(thread_idx)
             self.assertTrue(thread.IsValid())
             thread_id = thread.GetThreadID()
             self.assertIn(thread_id, expected_threads)
+            frame = thread.GetFrameAtIndex(0)
+            sp_region = lldb.SBMemoryRegionInfo()
+            sp = frame.GetSP()
+            err = process.GetMemoryRegionInfo(sp, sp_region)
+            self.assertTrue(err.Success(), err.GetCString())
+            error = lldb.SBError()
+            # Try to read at the end of the stack red zone and succeed
+            process.ReadMemory(sp_region.GetRegionEnd() - 1, 1, error)
+            self.assertTrue(error.Success(), error.GetCString())
+            # Try to read just past the red zone and fail
+            process.ReadMemory(sp_region.GetRegionEnd() + 1, 1, error)
+            # Try to read from the base of the stack
```

Comment:
1. For checking top of the stack + red zone, isn't it GetRegionBase() because stack grows toward lower address?
2. This function is used by not only stack only option but all options so I am not sure implementation details of stack-only dump uses red zone making sense here.
3. Even for stack-only dump, I am not sure this is testing anything related with red zone, it simply tests that dump does not contain memory outside its memory region...

Reply:
Fixed this, it was due to a misunderstanding on my part. We want to ensure we cannot read past the bottom of the truncated stack. I originally tried to do this with memory ranges, but actually I needed to do `sp - red_zone - 1` to then exceed the saved memory.